Polish the `MultiImage` docstring.

mkcor opened the pull request:
## Description

This minor PR builds on top of #6290 by @thvoigtmann (sorry, I couldn't review yesterday). This changeset is only intended to make the wording more standard and more direct. Notably,

* I removed "in the current implementation" because there is no reason to imply anything about a future implementation;
* since the object created by `ImageCollection` is named `ic`, I thought the one created by `MultiImage` might as well be named `mi` (rather than `img`).


## Checklist

<!-- It's fine to submit PRs which are a work in progress! -->
<!-- But before they are merged, all PRs should provide: -->
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

<!-- For detailed information on these and other aspects see -->
<!-- the scikit-image contribution guidelines. -->
<!-- https://scikit-image.org/docs/dev/contribute.html -->

## For reviewers

<!-- Don't remove the checklist below. -->
- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

mkcor commented on skimage/io/collection.py:
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
-    For an animated GIF image, `MultiImage` in the current implementation
-    will only read one frame, while `ImageCollection` by default will read
-    all of them.
+    For an animated GIF image, `MultiImage` reads only the first frame, while
```
That's what I noticed with `no_time_for_that_tiny.gif` -- hoping it's not any random one that `MultiImage` loads...

thvoigtmann replied:
@mkcor for the GIFs that I have tried, `MultiImage` indeed seems to load the first frame. At the moment I am not even sure `ImageCollection` does the right thing for animated GIF. Could it be that it detects all the frames but always returns the first one? I will try to look into this in the context of [#6291](https://github.com/scikit-image/scikit-image/pull/6291) once I have time.

grlee77 reviewed (approved):
The doc fixes here look good.

It seems there is still some question about the expected behavior for animated GIFs, though, so let's wait a bit before merge to see if that can be resolved. Alternatively, if a separate issue is opened for that we can just merge this now.

alexdesiqueira commented on skimage/io/collection.py:
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
-    where each entry in the list represents one image. In this regard it is
-    very similar to `ImageCollection`, but the two differ in the treatment
-    of multi-frame images.
+    The object that is returned is a list of image-data arrays. In this
```
What if:
```suggestion
    `MultiImage` returns a list of image-data arrays. In this
```
Then the other sentence (`In this regard, it is...`) would make more sense, I guess.

mkcor replied:
You're absolutely right, thank you!

alexdesiqueira commented on skimage/io/collection.py:
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
-    all frames of that image as a single entry of shape `(N, W, H)` in the
-    list. `ImageCollection` instead creates N entries of shape `(W, H)`.
+    all frames of that image as a single element of shape `(N, W, H)` in the
+    list. `ImageCollection` instead creates N elements of shape `(W, H)`.
```
```suggestion
    list. `ImageCollection`, in its turn, creates N elements of shape `(W, H)`.
```

mkcor replied:
I don't think so... Your wording would suggest that `ImageCollection` is called at some point.

alexdesiqueira commented on skimage/io/collection.py:
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
-    will only read one frame, while `ImageCollection` by default will read
-    all of them.
+    For an animated GIF image, `MultiImage` reads only the first frame, while
+    `ImageCollection` reads all of them by default.
```
```suggestion
    `ImageCollection` reads all frames by default.
```

mkcor replied:
Improves clarity indeed, thanks.

alexdesiqueira commented on skimage/io/collection.py:
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
     >>> multipage_tiff = data_dir + '/multipage.tif'
-    >>> img = MultiImage(multipage_tiff)
-    >>> len(img) # img contains one file
+    >>> mi = MultiImage(multipage_tiff)
```
I'm personally not very fond of `mi` here. How about something more descriptive? :slightly_smiling_face: 
```suggestion
    >>> multi_img = MultiImage(multipage_tiff)
```

mkcor replied:
Ok!

alexdesiqueira commented on skimage/io/collection.py:
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
-    >>> img = MultiImage(multipage_tiff)
-    >>> len(img) # img contains one file
+    >>> mi = MultiImage(multipage_tiff)
+    >>> len(mi)  # mi contains one element
```
If you agree with the previous comment,
```suggestion
    >>> len(multi_img)  # multi_img contains one element
```

alexdesiqueira commented on skimage/io/collection.py:
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
+    >>> len(mi)  # mi contains one element
     1
-    >>> img[0].shape # this image contains two frames of size (15, 10)
+    >>> mi[0].shape  # this element is a two-frame image of shape (2, 15, 10)
```
```suggestion
    >>> multi_img[0].shape  # this element is a two-frame image of shape (2, 15, 10)
```

alexdesiqueira commented on skimage/io/collection.py:
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
+    >>> mi[0].shape  # this element is a two-frame image of shape (2, 15, 10)
     (2, 15, 10)
 
     >>> ic = ImageCollection(multipage_tiff)
```
I'd say the same here. How about something more descriptive?
```suggestion
    >>> image_col = ImageCollection(multipage_tiff)
```
I know lots of our documentation have `ic` around, but I'll get there eventually :slightly_smiling_face:

mkcor replied:
I agree with your suggestions, I think they make the documentation more readable :relaxed:

alexdesiqueira commented on skimage/io/collection.py:
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
 
     >>> ic = ImageCollection(multipage_tiff)
-    >>> len(ic) # ic contains two images
+    >>> len(ic)  # ic contains two elements
```
If...
```suggestion
    >>> len(image_col)  # image_col contains two elements
```

alexdesiqueira commented on skimage/io/collection.py:
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
-    >>> len(ic) # ic contains two images
+    >>> len(ic)  # ic contains two elements
     2
     >>> for frame in ic:
```
```suggestion
    >>> for frame in image_col:
```

alexdesiqueira reviewed (changes requested):
Thank you @mkcor @thvoigtmann! I've suggested a few things that improve readability in my mind; please feel free to agree with or discuss them.

alexdesiqueira commented:
@grlee77 could you check if the changes I requested make sense, and add them if they do? :slightly_smiling_face: I think this one could be nicely merged after that. Thank you!

thvoigtmann commented:
@alexdesiqueira @mkcor Thanks for the updates, looking good to me.

@grlee77 Just to clarify regarding the animated GIFs, I think the purpose here is just to match the docstring with what actually happens. (Here: only the first frame of the GIF is loaded.) If that is "expected behavior" can be debated :-)

I'm happy to have you all commenting on [PR #6291](https://github.com/scikit-image/scikit-image/pull/6291) where I think I've managed to resolve the issue. Please see the "Updated Description" there. This is still marked as "draft" because I am a bit short on coding time these days and there is still a failing test that I don't understand.

pep8speaks commented:
Hello @mkcor! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`skimage/io/collection.py`](https://github.com/scikit-image/scikit-image/blob/4e5b0ff80900f2c7aa3aa48be17663addfc07fba/skimage/io/collection.py):

> [Line 438:80](https://github.com/scikit-image/scikit-image/blob/4e5b0ff80900f2c7aa3aa48be17663addfc07fba/skimage/io/collection.py#L438): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)

mkcor commented:
Hi @alexdesiqueira,

Thank you so much for the detailed review! I have finally addressed your suggestions. When you squash and merge, feel free to collapse the last commit messages (me not having updated my local setup yet :roll_eyes:).

alexdesiqueira reviewed (approved):
I'm happy. Thank you @mkcor!

thvoigtmann commented:
Oh wow, a burst of activity today. Thank you :)
